CorsixTH/Lua/entities/vip.lua

```diff
@@ -294,26 +294,31 @@ function Vip:setVIPRating()
   end
 
   -- check for the average queue length
-  local sum_queue = 0
-  for _, room in pairs(self.world.rooms) do
-    -- this can be nil if there has been a room explosion
-    if room.door.queue then
-      sum_queue = sum_queue + room.door.queue:size()
+  if #self.world.rooms ~= 0 then
+    local sum_queue = 0
+    local room_count = 0
+    for _, room in pairs(self.world.rooms) do
+      if room.door.queue then
+        sum_queue = sum_queue + room.door.queue:size()
+        room_count = room_count + 1
+      end
```

Comment:
The game seems to use `room.door.queue` as "this is a valid room" condition rather than `#self.world.rooms ~= 0`. In other words, your change opens the option that there are rooms in the world, but none has a queue, and thus none is counted.
Wouldn't it be simpler to keep this counting part as it was, and instead use `room_count` in the vip-rating above your new line 307 (if not rooms -> -100, elif no queue -> +6, else -> complicated calculation) ?

Reply:
It would, reverted back to an earlier version of the loop.